### PR TITLE
chore: update dependencies and ci

### PR DESCRIPTION
**Pull request by bhelx**

### CODE REVIEW

File: crates/cli/src/opt.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ impl<'a> Optimizer<'a> {
         let mut wasm = Wizer::new()
             .allow_wasi(true)?
             .inherit_stdio(true)
+            .wasm_bulk_memory(true)
```
Comment (bhelx) [inline crates/cli/src/opt.rs:27]: needed for newer rust versions

File: crates/cli/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ anyhow = { workspace = true }
 wasm-encoder = "0.20.0"
 wasmparser = "0.96.0"
 parity-wasm = { version = "0.45.0", features = ["bulk"] }
-wizer = "1.6.0"
+wizer = "^3.0.0"
```
Comment (bhelx) [inline crates/cli/Cargo.toml:18]: none of the breaking changes have an effect on us

**Review by wikiwong — APPROVED** [review]
lgtm

**Comment by bhelx** [comment]
One of the opcodes in the wasm file in the test was `0xc0` which suggested the sign-extension features are needed. Probably because I'm using a modern version of the rust compiler that emits it but parity-wasm is getting a bit old so i had to enable it as a feature.

**Pull request merged**